api/src/main/java/com/velocitypowered/api/network/ProtocolVersion.java

```diff
@@ -0,0 +1,166 @@
+package com.velocitypowered.api.network;
+
+import com.google.common.collect.ImmutableMap;
+import com.velocitypowered.api.proxy.server.ServerInfo;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Represents each Minecraft protocol version
+ */
+public enum ProtocolVersion {
+    UNKNOWN(-1, "Unknown"),
+    LEGACY(-1, "Legacy"),
+    MINECRAFT_1_8(47, "1.8"),
+    MINECRAFT_1_9(107, "1.9"),
+    MINECRAFT_1_9_1(108, "1.9.1"),
+    MINECRAFT_1_9_2(109, "1.9.2"),
+    MINECRAFT_1_9_4(110, "1.9.4"),
+    MINECRAFT_1_10(210, "1.10"),
+    MINECRAFT_1_11(315, "1.11"),
+    MINECRAFT_1_11_1(316, "1.11.1"),
+    MINECRAFT_1_12(335, "1.12"),
+    MINECRAFT_1_12_1(338, "1.12.1"),
+    MINECRAFT_1_12_2(340, "1.12.2"),
+    MINECRAFT_1_13(393, "1.13"),
+    MINECRAFT_1_13_1(401, "1.13.1"),
+    MINECRAFT_1_13_2(404, "1.13.2");
+
+    private final int protocol;
+    private final String name;
+
+    public static final ProtocolVersion MINIMUM_GENERIC_VERSION = MINECRAFT_1_8;
```

Comment:
Perhaps we ought to drop the `GENERIC`?

Reply:
Drop `GENERIC` from the `SUPPORTED_GENERIC_VERSION_STRING` also?

Reply:
Dropped it anyways to conform to the `MINIMUM` and `MAXIMUM` variables